### PR TITLE
Update build to remove UTF-8 chars.

### PR DESCRIPTION
**Pull request by bfMega**

### CODE REVIEW

**Review by diegocr — CHANGES_REQUESTED** [review]
Thanks Ben!

Under https://github.com/meganz/Autolinker.js/commit/e63dd8c0016e503e17b05f08ff97c2ba5417bf1f i took the chance to switch the build procedure to generate ES2018-compliant files instead.

Would you mind adding into `scripts/build.ts` a function that programmatically changes the UTF-8 characters, as a mean to ease further maintenance / upgrades in the long run, please? If you do find any trouble with this just sync the changes and apply manually, tho.

**Comment by bfMega** [comment]
Removed some of the problematic characters in the rest of the code + Updated the tldRegex generating script to replace the UTF-8 characters. Also changed the rollup config to match the filename we use.

File: scripts/update-tld-regex.ts [hunk 1]
```diff
@@ -16,7 +16,15 @@ export async function updateTldRegex() {
         responseType: 'text',
     });
 
-    const tldRegex = domainsToRegex(tldsFile.data);
+    let tldRegex = domainsToRegex(tldsFile.data);
+    const matches = tldRegex.match(/[\u0100-\uFFFF]/g) || [];
```
Comment (diegocr) [inline scripts/update-tld-regex.ts]:
```suggestion
    const matches = tldRegex.match(/[\u007F-\uFFFF]/g) || [];
```

File: scripts/update-tld-regex.ts [hunk 2]
```diff
@@ -16,7 +16,15 @@ export async function updateTldRegex() {
...
+    matches.forEach(match => {
+        let codePoint = (match.codePointAt(0) || 0).toString(16);
+        if (codePoint.length < 4) {
+            codePoint = `0${codePoint}`;
+        }
```
Comment (diegocr) [inline scripts/update-tld-regex.ts]:
```suggestion
        const codePoint = (match.codePointAt(0) | 0).toString(16).padStart(4, '0');
```

**Comment by diegocr** [comment]
Overall looks good, will take care myself from there, thanks!

**Review by diegocr — APPROVED** [review]

**Pull request merged**